src/kroneckersum.jl

```diff
@@ -59,6 +59,15 @@ documentation.
 ⊕(A::AbstractMatrix...) = kroneckersum(A...)
 ⊕(A::AbstractMatrix, pow::Int) = kroneckersum(A, pow)
 
+
+"""
+    getallmatrices(K::T) where T <: AbstractKroneckerSum
+
+Obtain the all matrices of an `AbstractKroneckerSum` object.
+"""
+Kronecker.getallmatrices(K::AbstractKroneckerSum) = (Kronecker.getallmatrices(K.A)..., Kronecker.getallmatrices(K.B)...)
```

Comment:
I'm guessing you want to use this implementation _exclusively_ for fast multiplications with Kronecker sums 
(in `mul!(out::AbstractVecOrMat, K::AbstractKroneckerSum, x::AbstractVecOrMat`) . :heart_eyes:

Suppose you have the higher order products/sums:
```julia
K1 = A ⊗ B ⊗ C
K2 = A ⊕ B ⊕ C
```
Currently, you would get `getallmatrices(K1) == getallmatrices(K2) == (A, B, C)` and run the risk of forgetting 
 what the original subtype of  the `GeneralizedKroneckerProduct` you are working with.

Although `getallmatrices` isn't exported, this behavior may need to be documented if someone wants to use it themselves.

I imagine that this would be a problem with dispatching `mul!` for (sum)mixed products like `A ⊕ B ⊗ C ⊕ D`, but that case may be outside the scope of this PR.

Reply:
yeah that is a problem...

i'll change it to `getallfactors` for products/powers and `getallsummands` for kronsums, unless you have any better names?

Reply:
Maybe shorten each to `getfactors` and `getsummands`?

Reply:
i think those could end up being confused with `getmatrices` in that users may expect it to return a 2-tuple